src/utils/query_builder.py

```diff
@@ -130,7 +130,7 @@ def add_sql_group_by_condition(self, columns: list):
             raise Exception("No valid columns for group by")
 
     def add_sql_limit_condition(self, limit):
-        if self.__is_valid_number(limit):
+        if self.__is_valid_number(limit) or limit is None:
```

Comment:
Could you add condition in __build_offset function to verify limit is not None also?